nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -150,548 +145,320 @@
                     }
-
-                    while (true) {
-                        if (headerInfoLineCount > 0) {
-                            // if we have header lines, create a new FlowFile, copy the header lines to that file,
-                            // and then start copying lines
-                            final AtomicInteger linesCopied = new AtomicInteger(0);
-                            final AtomicLong bytesCopied = new AtomicLong(0L);
-                            FlowFile splitFile = session.create(flowFile);
-                            try {
-                                splitFile = session.write(splitFile, new OutputStreamCallback() {
-                                    @Override
-                                    public void process(final OutputStream rawOut) throws IOException {
-                                        try (final BufferedOutputStream out = new BufferedOutputStream(rawOut);
-                                                final ByteCountingOutputStream countingOut = new ByteCountingOutputStream(out)) {
-                                            countingOut.write(headerBytesWithoutTrailingNewLines);
-                                            //readLines has an offset of countingOut.getBytesWritten() to allow for header bytes written already
-                                            linesCopied.set(readLines(in, maxLineCount, maxFragmentSize - countingOut.getBytesWritten(), countingOut,
-                                                    includeLineDelimiter, headerNewLineBytes));
-                                            bytesCopied.set(countingOut.getBytesWritten());
-                                        }
-                                    }
-                                });
-                                splitFile = session.putAttribute(splitFile, SPLIT_LINE_COUNT, String.valueOf(linesCopied.get()));
-                                splitFile = session.putAttribute(splitFile, FRAGMENT_SIZE, String.valueOf(bytesCopied.get()));
-                                logger.debug("Created Split File {} with {} lines, {} bytes", new Object[]{splitFile, linesCopied.get(), bytesCopied.get()});
-                            } finally {
-                                if (linesCopied.get() > 0) {
-                                    splits.add(splitFile);
+                }
+            });
+            if (!error.get()) {
+                FlowFile headerFlowFile = null;
+                long headerCrlfLength = 0;
+                if (headerSplitInfoRef.get() != null) {
+                    headerFlowFile = session.clone(flowFile, headerSplitInfoRef.get().startOffset, headerSplitInfoRef.get().length);
+                    headerCrlfLength = headerSplitInfoRef.get().trimmedLength;
+                }
+                int fragmentIndex = 1; // set to 1 to preserve the existing behavior *only*. Perhaps should be deprecated to follow the 0,1,2... scheme
```

Comment:
I believe that leaving this as 1 is correct, and that it should not be changed to 0. Changing this would break backward compatibility and provides no real benefit. Though 0 is a more common starting point for developers, 1 is also used commonly enough, and 1 was used as the starting point here specifically because if you look at FlowFile attributes and see, for example, a 3, it is more intuitive to think that this is the 3rd one in a series, rather than the 4th one.